### PR TITLE
Hotfix tpd docs

### PR DESCRIPTION
**Pull request by mbaker3**
Minor documentation improvements and calling out the pending work of #283 

### What is the current behaviour?
 - The meaning of "World Unique" is unclear
 - `ThreadPersistentData` has a misleading name. This is well described in #283 

### What is the new behaviour?
 - Clarified that "World Unique" means unique within the world instance
 - Add `//TODO: #283` callouts where appropriate

### What issues does this resolve?
 - None

### What PRs does this depend on?

 - None

### Does this introduce a breaking change?
 - [ ] Yes 
 - [x] No

### CODE REVIEW

File: Scripts/Runtime/Entities/Data/IWorldUniqueID.cs [hunk 1]
```diff
@@ -3,13 +3,17 @@
 namespace Anvil.Unity.DOTS.Entities
 {
     /// <summary>
-    /// Provides a <see cref="World"/> unique ID
+    /// Provides a <see cref="World"/> unique ID.
+    /// This ID is unique within the current World instance.
+    /// This ID is not unique globally between all world instances.
```
Comment (jkeon) [inline Scripts/Runtime/Entities/Data/IWorldUniqueID.cs:8]: I would be more explicit here:

```suggestion
    /// This ID is guaranteed to be the same across all world instances.
```

File: Scripts/Runtime/Entities/Data/IWorldUniqueID.cs [hunk 2]
```diff
@@ -3,13 +3,17 @@
...
     /// </summary>
     /// <typeparam name="TWorldUniqueID">The type of ID</typeparam>
     public interface IWorldUniqueID<out TWorldUniqueID>
     {
         /// <summary>
-        /// The <see cref="World"/> unique ID
+        /// The <see cref="World"/> unique ID.
+        /// This ID is unique within the current World instance.
+        /// This ID is not unique globally between all world instances.
```
Comment (jkeon) [inline Scripts/Runtime/Entities/Data/IWorldUniqueID.cs:16]: Same thing here

**Review by jkeon — DISMISSED** [review]
Looks good, one suggestion for improvement.

File: Scripts/Runtime/Entities/Data/IWorldUniqueID.cs (hunk 2 above)
Comment (mbaker3) [reply Scripts/Runtime/Entities/Data/IWorldUniqueID.cs]: fixed

**Review by jkeon — APPROVED** [review]

**Pull request merged**